proto-lens-tests/tests/decode_delimited.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main (main) where
+
+import qualified Data.ByteString as B
+import Data.ProtoLens
+import Data.ProtoLens.Encoding.Bytes (runBuilder)
+import Lens.Family2 ((&), (.~))
+import Test.Framework.Providers.HUnit (testCase)
+import Test.HUnit ((@=?))
+
+import Data.ProtoLens.TestUtil
+import Proto.DecodeDelimited
+import Proto.DecodeDelimited_Fields
+
+import System.IO (openBinaryFile, hClose, IOMode(ReadMode))
+import System.Directory (removeFile)
+import Control.Exception (catch, throwIO)
+import System.IO.Error (isDoesNotExistError)
+
+filename :: String
+filename = "test_decode_delimited.dat"
+
+removeIfExists :: FilePath -> IO ()
+removeIfExists fileName = removeFile fileName `catch` handleExists
+  where handleExists e
+          | isDoesNotExistError e = return ()
+          | otherwise = throwIO e
+
+main :: IO ()
+main = testMain
+    [ testCase "buildDelimited/decodeDelimited" $ do
+          let foo = defMessage & a .~ 42 & b .~ "hello" :: Foo
+          let bs = runBuilder . buildMessageDelimited $ foo
+          B.writeFile filename bs
```

Comment:
Let's use the `temporary` package to avoid writing files in the current directory, which would be the `proto-lens-tests` source directory in this case.
http://hackage.haskell.org/package/temporary-1.3/docs/System-IO-Temp.html#v:withSystemTempFile